src/veh_interact.cpp

```diff
@@ -934,79 +928,180 @@ void veh_interact::do_mend()
+            auto details = []( const vehicle_part &pt, WINDOW *w, int y ) {
+                int pct = ( double( pt.ammo_remaining() ) / pt.ammo_capacity() ) * 100;
+                right_print( w, y, 1, c_yellow, "%i    %3i%%", pt.ammo_capacity(), pct );
+            };
+           opts.push_back( part_option { "BATTERY", &pt, enable ? enable( pt ) : false, details } );
         }
-    };
+    }
 
-    if (ptanks.empty()) {
-        debugmsg("veh_interact::do_refill: Refillable parts list is empty.\n"
-                 "veh_interact::cant_do() should control this before.");
-        return;
+    for( auto &pt : veh->parts ) {
+        if( pt.is_reactor() && !pt.is_broken() ) {
+            // if reactor is fuelled then display contents in scaled plutonium units
+            auto details = []( const vehicle_part &pt, WINDOW *w, int y ) {
+                if( pt.ammo_remaining() ) {
+                    right_print( w, y, 1, item::find_type( pt.ammo_current() )->color,
+                                 "%s    %4i", item::nname( pt.ammo_current() ).c_str(), pt.ammo_remaining() );
+                }
+            };
+            opts.push_back( part_option { "REACTOR", &pt, enable ? enable( pt ) : false, details } );
+        }
+    }
+
+    for( auto &pt : veh->parts ) {
+        if( pt.is_turret() && !pt.is_broken() ) {
+            // if turret is loaded then display details about contained ammo
+            auto details = [&]( const vehicle_part &pt, WINDOW *w, int y ) {
+                if( pt.ammo_current() != "null" ) {
+                    right_print( w, y, 1, item::find_type( pt.ammo_current() )->color,
+                                 "%s   %5i", item::nname( pt.ammo_current() ).c_str(),
+                                 pt.ammo_remaining() );
+                }
+            };
+            opts.push_back( part_option { "TURRET", &pt, enable ? enable( pt ) : false, details } );
+        }
     }
-    // Now at least one of "fuel tank" is can be refilled.
-    // If we have more that one tank we need to create choosing menu
-    if (ptanks.size() > 1) {
-        unsigned int pt_choise;
-        unsigned int entry_num;
-        uimenu fuel_choose;
-        fuel_choose.text = _("What to refill:");
-        for( entry_num = 0; entry_num < ptanks.size(); entry_num++) {
-            const vpart_info &vpinfo = ptanks[entry_num]->info();
-            fuel_choose.addentry(entry_num, true, -1, "%s -> %s",
-                                 item::nname(vpinfo.fuel_type).c_str(),
-                                 ptanks[ entry_num ]->name().c_str());
-        }
-        fuel_choose.addentry(entry_num, true, 'q', _("Cancel"));
-        fuel_choose.query();
-        pt_choise = fuel_choose.ret;
-        if(pt_choise == entry_num) { // Select canceled
+
+    int pos = enable && action ? 0 : -1;
+    while( true ) {
+        werase( w_list );
+        std::string last;
+        int y = 0;
+        for( int idx = 0; idx != int( opts.size() ); ++idx ) {
```

Comment:
Why not define `size_t idx `? It's a good index. `if( !enable || !action )` (below) together with `int pos = enable && action ? 0 : -1;` clearly mean that the condition is misplaced, and `pos` can become `size_t` too.